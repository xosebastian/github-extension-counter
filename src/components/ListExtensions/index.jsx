import * as React from 'react';
import PropTypes from 'prop-types';
import ListItem from '@mui/material/ListItem';
import ListItemText from '@mui/material/ListItemText';
import ListItemAvatar from '@mui/material/ListItemAvatar';
import Avatar from '@mui/material/Avatar';
import Grid from '@mui/material/Grid';
import Box from '@mui/material/Box';

function ListExtensions({ data }) {
  return (
    <Box sx={{ flexGrow: 1, mt: 2 }}>
      <Grid container spacing={0}>
        {data.map(({ name, count }) => (
          <Grid item md={4} key={name}>
            <ListItem>
              <ListItemAvatar>
                <Avatar>
                  {name.toUpperCase().charAt(0, 1)}
                  {' '}
                </Avatar>
              </ListItemAvatar>
              <ListItemText
                primary={`Extension: ${name.toUpperCase()}`}
                secondary={`Quantity: ${count}`}
              />
            </ListItem>
          </Grid>
        ))}
      </Grid>
    </Box>
  );
}

ListExtensions.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape(
      {
        name: PropTypes.string.isRequired,
        count: PropTypes.number.isRequired,
      }.isRequired,
    ).isRequired,
  ).isRequired,
};

export default ListExtensions;
