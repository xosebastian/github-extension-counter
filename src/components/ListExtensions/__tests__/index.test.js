/**
 * @jest-environment jsdom
 */

import React from "react";
import { render } from "@testing-library/react";
import ListExtensions from "../index";

const data = [
  {
    name: "js",
    counst: 2,
  },
];

const getRenderedComponent = () => render(<ListExtensions data={data} />);

describe("<ListExtensions/>", () => {
  it("Must render the component", () => {
    const { baseElement } = getRenderedComponent();
    expect(baseElement).toMatchSnapshot();
  });

  it("Expect to not log errors in console", () => {
    const spy = jest.spyOn(global.console, "error");
    getRenderedComponent();
    expect(spy).not.toHaveBeenCalled();
  });
});
