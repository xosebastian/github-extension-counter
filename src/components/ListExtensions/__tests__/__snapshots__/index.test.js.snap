// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListExtensions/> Must render the component 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-be40y1"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-4 css-1ube6tp-MuiGrid-root"
        >
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
          >
            <div
              class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
              >
                J
                 
              </div>
            </div>
            <div
              class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
              >
                Extension: JS
              </span>
              <p
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
              >
                Quantity: undefined
              </p>
            </div>
          </li>
        </div>
      </div>
    </div>
  </div>
</body>
`;
